### PR TITLE
Update MPs functionality

### LINKED ISSUE #297: Tweak MP fields

### PR DESCRIPTION
**Pull request by AndyEPhipps**
Fixes #297

### CODE REVIEW

**Comment by AndyEPhipps** [comment]
@KrupaPammi : as per our discussions on Slack, I've added in options to show copy per preference (and associated opt in or out), to better explain the functionality:
Screen Shot 2021-05-06 at 12 14 40

I'm going to run the exact copy past Ruby, but I feel like this makes everything clearer now :)

I've also updated the placeholder for Telephone to make it less specific to either mobile or landline.

File: src/pages/GiftAid/utils/Utils.js [hunk 1]
```diff
@@ -76,9 +76,10 @@ export const getFormValues = (validation, urlId = null, update = false) => {
     if (key === 'giftAidClaimChoice') {
       fieldValues.confirm = parseInt(value); // reassign to confirm field
     }
+
     // set values for marketing consent checkboxes and fields
     if (/^permission/.test(key) && value !== null) {
-      if (value === 'yes' && validation[key].fieldValidation !== false) {
+      if (validation[key].fieldValidation !== false) {
```
Comment (AndyEPhipps) [inline src/pages/GiftAid/utils/Utils.js:82]: This changes allows us to submit for 'no' values too, which is required for an opt OUT

**Review by gusliedke — APPROVED** [review]

**Comment by AndyEPhipps** [comment]
@KrupaPammi just an aside that the old Storybook input and MP components (that GA uses) uses some weird old validation rules; not a proper phone number check, etc.

I'm in the middle of updating the newer Component Library version of MPs to use the Yup validation rules we're using everywhere, but I'll now see if i can do the same within the Storybook, but will actually add it to GA in a separate PR.

**Comment by AndyEPhipps** [comment]
That's great, thanks for that @KrupaPammi. 

Issue 2 _is_ an existing bug, but will be addressed by me finishing up that proper Yup validation rule stuff I mentioned above, which I'd like to keep in a separate PR if possible. I'll see if I can can address issue 1 in that same PR, but I can see it being very fiddly :/ Just weary of burning time finessing some already very old code!

**Comment by AndyEPhipps** [comment]
Further to our standup chat, are you happy for me to the merge this in ahead of your test updates, @KrupaPammi ?

**Comment by KrupaPammi** [comment]
@AndyEPhipps  yes, happy for this to be merged in. I have to modify the tests so might take time.

**Comment by AndyEPhipps** [comment]
No rush at all @KrupaPammi :)

**Pull request merged**